### PR TITLE
Mamba enablement

### PR DESCRIPTION
**Pull request by libinta**
# What does this PR do?
Initial emblement for mamba with static shape.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: optimum/habana/transformers/models/mamba/modeling_mamba.py
Comment (regisss) [inline optimum/habana/transformers/models/mamba/modeling_mamba.py:1]: Please put the methods in the same order as they appear in the Transformers modeling file

**Review by regisss — COMMENTED** [review]
Let's add a test in `test_text_generation.py` too no?

File: optimum/habana/transformers/models/mamba/modeling_mamba.py
Comment (MohitIntel) [reply optimum/habana/transformers/models/mamba/modeling_mamba.py:1]: Done in latest commit.

File: tests/resource/image-captioning-example.png
Comment (MohitIntel) [inline tests/resource/image-captioning-example.png:1]: this file was moved to img folder in the latest commit on trasnformers_4.40 branch.

**Comment by MohitIntel** [comment]
> Let's add a test in `test_text_generation.py` too no?

Added in latest commit.

File: optimum/habana/transformers/modeling_utils.py [hunk 1]
```diff
@@ -438,3 +440,11 @@ def adapt_transformers_to_gaudi():
 
     # Tell transformers which Gaudi models support tracing
     transformers.utils.fx._SUPPORTED_MODELS += tuple(cls.__name__ for cls in models_with_tracing_support)
+
+    # Optimization for mamaba on Gaudi
```
Comment (mfuntowicz) [inline optimum/habana/transformers/modeling_utils.py:444]:
```suggestion
    # Optimization for mamba on Gaudi
```

**Review by mfuntowicz — APPROVED** [review]
LGTM

File: tests/resource/image-captioning-example.png
Comment (regisss) [inline tests/resource/image-captioning-example.png:1]: I don't think this is related to this PR no?

Comment (MohitIntel) [reply tests/resource/image-captioning-example.png:1]: Not related to this PR.  Git was not letting me push my branch otherwise.

Comment (regisss) [reply tests/resource/image-captioning-example.png:1]: Hmm that's weird. Is your branch up to date with `main`?

**Pull request closed without merging**